create_gazebo_plugins failure to compile with gazebo7

Gazebo 7 uses c++11 headers which means that this doesn't compile.  Either this needs to add c++11 flags to the objects using those headers or it should be ticketed upstream for gazebo_ros to export the flags. Though even if they are exported above, we'll need to use them in this package. 

There's a related discussion at: https://github.com/ros/resource_retriever/pull/7 which was resolved by
https://github.com/ros-visualization/rviz/commit/faf97becbbb51e08227e00557d171a9c60a81c79

http://build.ros.org:8080/view/Queue/job/Kbin_uX32__create_gazebo_plugins__ubuntu_xenial_i386__binary/1/console

```
15:27:00 [ 50%] Building CXX object CMakeFiles/gazebo_ros_create.dir/src/gazebo_ros_create.cpp.o
15:27:00 /usr/lib/ccache/i686-linux-gnu-g++   -DROSCONSOLE_BACKEND_LOG4CXX -DROS_PACKAGE_NAME=\"create_gazebo_plugins\" -Dgazebo_ros_create_EXPORTS -I/tmp/binarydeb/ros-kinetic-create-gazebo-plugins-2.3.1/include -I/opt/ros/kinetic/include -I/usr/include/gazebo-7 -I/usr/include/sdformat-4.0 -I/usr/include/ignition/math2  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2  -fPIC   -o CMakeFiles/gazebo_ros_create.dir/src/gazebo_ros_create.cpp.o -c /tmp/binarydeb/ros-kinetic-create-gazebo-plugins-2.3.1/src/gazebo_ros_create.cpp
15:27:07 In file included from /usr/include/c++/5/random:35:0,
15:27:07                  from /usr/include/ignition/math2/ignition/math/Rand.hh:20,
15:27:07                  from /usr/include/ignition/math2/ignition/math.hh:18,
15:27:07                  from /usr/include/sdformat-4.0/sdf/Param.hh:34,
15:27:07                  from /usr/include/sdformat-4.0/sdf/Element.hh:24,
15:27:07                  from /usr/include/sdformat-4.0/sdf/sdf.hh:5,
15:27:07                  from /usr/include/gazebo-7/gazebo/sensors/SensorManager.hh:26,
15:27:07                  from /tmp/binarydeb/ros-kinetic-create-gazebo-plugins-2.3.1/src/gazebo_ros_create.cpp:1:
15:27:07 /usr/include/c++/5/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
15:27:07  #error This file requires compiler and library support \
15:27:07   ^
15:27:10 In file included from /usr/include/sdformat-4.0/sdf/sdf.hh:3:0,
15:27:10                  from /usr/include/gazebo-7/gazebo/sensors/SensorManager.hh:26,
15:27:10                  from /tmp/binarydeb/ros-kinetic-create-gazebo-plugins-2.3.1/src/gazebo_ros_create.cpp:1:
```

--- a/create_gazebo_plugins/CMakeLists.txt
+++ b/create_gazebo_plugins/CMakeLists.txt
@@ -7,6 +7,8 @@ if (PKG_CONFIG_FOUND)
   pkg_check_modules(GAZEBO gazebo)
 endif()
 
+set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} " -std=c++11")
+
 find_package(catkin REQUIRED COMPONENTS create_node
                                         gazebo_ros
                                         geometry_msgs

--- a/create_gazebo_plugins/src/gazebo_ros_create.cpp
+++ b/create_gazebo_plugins/src/gazebo_ros_create.cpp
@@ -252,12 +252,12 @@ void GazeboRosCreate::UpdateChild()
     d2 = step_time.Double() * (wd / 2) * joints_[RIGHT]->GetVelocity(0);
 
   // Can see NaN values here, just zero them out if needed
-  if (isnan(d1)) {
+  if (std::isnan(d1)) {
     ROS_WARN_THROTTLE(0.1, "Gazebo ROS Create plugin. NaN in d1. Step time: %.2f. WD: %.2f. Velocity: %.2f", step_time.Double(), wd, joints_[LEFT]->GetVelocity(0));
     d1 = 0;
   }
 
-  if (isnan(d2)) {
+  if (std::isnan(d2)) {
     ROS_WARN_THROTTLE(0.1, "Gazebo ROS Create plugin. NaN in d2. Step time: %.2f. WD: %.2f. Velocity: %.2f", step_time.Double(), wd, joints_[RIGHT]->GetVelocity(0));
     d2 = 0;
   }

--- a/create_gazebo_plugins/src/gazebo_ros_create.cpp
+++ b/create_gazebo_plugins/src/gazebo_ros_create.cpp
@@ -110,7 +110,7 @@ void GazeboRosCreate::Load(physics::ModelPtr _parent, sdf::ElementPtr _sdf )
   std::string topic = mgr->CreateFilter(base_geom_name_, base_geom_name_);
   contact_sub_ = gazebo_node_->Subscribe(topic, &GazeboRosCreate::OnContact, this);
 
-  wall_sensor_ = boost::dynamic_pointer_cast<sensors::RaySensor>(
+  wall_sensor_ = std::dynamic_pointer_cast<sensors::RaySensor>(
     sensors::SensorManager::Instance()->GetSensor("wall_sensor"));
   if (!wall_sensor_)
   {
@@ -118,13 +118,13 @@ void GazeboRosCreate::Load(physics::ModelPtr _parent, sdf::ElementPtr _sdf )
     return;
   }
 
-  left_cliff_sensor_ = boost::dynamic_pointer_cast<sensors::RaySensor>(
+  left_cliff_sensor_ = std::dynamic_pointer_cast<sensors::RaySensor>(
     sensors::SensorManager::Instance()->GetSensor("left_cliff_sensor"));
-  right_cliff_sensor_ = boost::dynamic_pointer_cast<sensors::RaySensor>(
+  right_cliff_sensor_ = std::dynamic_pointer_cast<sensors::RaySensor>(
     sensors::SensorManager::Instance()->GetSensor("right_cliff_sensor"));
-  leftfront_cliff_sensor_ = boost::dynamic_pointer_cast<sensors::RaySensor>(
+  leftfront_cliff_sensor_ = std::dynamic_pointer_cast<sensors::RaySensor>(
     sensors::SensorManager::Instance()->GetSensor("leftfront_cliff_sensor"));
-  rightfront_cliff_sensor_ = boost::dynamic_pointer_cast<sensors::RaySensor>(
+  rightfront_cliff_sensor_ = std::dynamic_pointer_cast<sensors::RaySensor>(
     sensors::SensorManager::Instance()->GetSensor("rightfront_cliff_sensor"));
 
   wall_sensor_->SetActive(true);
@@ -279,12 +279,12 @@ void GazeboRosCreate::UpdateChild()
   if (set_joints_[LEFT])
   {
     joints_[LEFT]->SetVelocity( 0, wheel_speed_[LEFT] / (wd/2.0) );
-    joints_[LEFT]->SetMaxForce( 0, torque_ );
+    joints_[LEFT]->SetParam("max_force", 0, torque_);
   }
   if (set_joints_[RIGHT])
   {
     joints_[RIGHT]->SetVelocity( 0, wheel_speed_[RIGHT] / (wd / 2.0) );
-    joints_[RIGHT]->SetMaxForce( 0, torque_ );
+    joints_[RIGHT]->SetParam("max_force", 0, torque_);
   }
 
   nav_msgs::Odometry odom;

--- a/create_gazebo_plugins/src/gazebo_ros_create.cpp
+++ b/create_gazebo_plugins/src/gazebo_ros_create.cpp
@@ -279,12 +279,12 @@ void GazeboRosCreate::UpdateChild()
   if (set_joints_[LEFT])
   {
     joints_[LEFT]->SetVelocity( 0, wheel_speed_[LEFT] / (wd/2.0) );
-    joints_[LEFT]->SetParam("max_force", 0, torque_);
+    joints_[LEFT]->SetParam("fmax", 0, torque_);
   }
   if (set_joints_[RIGHT])
   {
     joints_[RIGHT]->SetVelocity( 0, wheel_speed_[RIGHT] / (wd / 2.0) );
-    joints_[RIGHT]->SetParam("max_force", 0, torque_);
+    joints_[RIGHT]->SetParam("fmax", 0, torque_);
   }
 
   nav_msgs::Odometry odom;

